FacesofLondon/app/src/main/java/com/novoda/wallpaper/droidcon/LondonParallaxWallpaper.java

```diff
@@ -0,0 +1,68 @@
+package com.novoda.wallpaper.droidcon;
+
+import android.service.wallpaper.WallpaperService;
+import android.util.Log;
+
+import java.io.IOException;
+
+import net.rbgrn.android.glwallpaperservice.GLWallpaperService;
+
+/**
+ * Draws a wallpaper much like the regular wallpapers which slide behind
+ * the home screen with a parallax effect. The difference with this wallpaper
+ * is that it supports multiple images with different widths, for a much more
+ * intricate-looking parallax effect.
+ * <p/>
+ * The main thing is that it's configurable, so you can use any set of images.
+ */
+public class LondonParallaxWallpaper extends GLWallpaperService {
+
+    public static final String TAG = "LondonParallaxWallpaper";
+
+    private LondonParallaxWallpaperRenderer renderer;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+    }
+
+    @Override
+    public WallpaperService.Engine onCreateEngine() {
+        return new ParallaxEngine();
+    }
+
+    class ParallaxEngine extends GLEngine {
+
+        ParallaxEngine() {
+            super();
```

Comment:
Yes, `WallpaperService.Engine()` does a bunch of initialisations in the constructor